### ISSUE TITLE
Compatibility when working on a repo across Windows and MacOS machines

### ISSUE THREAD

**Issue by craigtonlian** [nus-cs2103-AY2324S1/forum#2]
When working on a shared Github repo, if our team members have different operating systems on their machines, will it lead to any conflict or compatibility issues on Git due to line endings?

I came across a related post below, but I would like to tap on the collective wisdom of all to verify if such a configuration is still required and relevant.

Link: https://stackoverflow.com/questions/33574435/git-when-used-with-mac-and-windows-simultaneously-gives-unnecessary-conflicts

**Comment by damithc** [comment]
Good question @craigtonlian 
As this is not urgent (because the team project coding is still a few weeks away), let's wait to see what others think about this.

**Comment by HugeNoob** [comment]
I haven't had any issues working across Windows and macOS with the same git repository. 

However, I have faced issues when accessing the same repository on Windows WSL (Ubuntu) as described in: https://stackoverflow.com/questions/53900839/git-repo-gives-contradictory-info-from-wsl-than-from-windows

TLDR: Difference in the way line endings are interpreted causes conflicts on every line.

Fixed the issue with `git config --global core.autocrlf true`. Alternatively, you can set the git config locally in the `.git` folder for the project.

**Comment by craigtonlian** [comment]
> Fixed the issue with `git config --global core.autocrlf true`. Alternatively, you can set the git config locally in the `.git` folder for the project.

I'm not very bright so please bear with my clarification.

Do you mean that if every team member applies `git config --global core.autocrlf true` in their local machine, then the team should no longer face conflicts arising due to line ending representations across different operating systems?

**Comment by HugeNoob** [comment]
> Do you mean that if every team member applies `git config --global core.autocrlf true` in their local machine, then the team should no longer face conflicts arising due to line ending representations across different operating systems?

You can read https://blog.chapagain.com.np/git-fix-differences-created-by-line-endings-in-files/ and https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration (search autocrlf) for more information on the differences.

TLDR: 
If you don't face any issues, then I guess it's fine to leave things as they are. Otherwise, you can do:  
`git config --global core.autocrlf true` on Windows
`git config --global core.autocrlf input` on macOS/Linux

I'm not 100% sure about whether everyone needs to do this on their own machine. I think it suffices if the person facing the conflicts does it. Would be good if more people can confirm.

**Comment by cheeheng** [comment]
Hi. I am a student from a previous batch. I would like to remind the original poster to close the issue if it has been resolved.

**Comment by craigtonlian** [comment]
> Hi. I am a student from a previous batch. I would like to remind the original poster to close the issue if it has been resolved.

Hi. The original poster appreciates the kind reminder, but he will only know if the issue has been resolved when tP starts a few weeks later. Thanks for being patient.

**Comment by craigtonlian** [comment]
No issues encountered so far in tP. Will close this issue.
